src/platform/string_traits.h

```diff
@@ -2,24 +2,20 @@
 
 #include <stdint.h>
 
-template <typename char_type>
-struct alternate_type;
-
-template <>
-struct alternate_type<xlang_char8>
+namespace xlang::impl
 {
-    using result_type = char16_t;
-};
+    template <typename char_type>
+    struct alternate_type;
 
-template <>
-struct alternate_type<char16_t>
-{
-    using result_type = xlang_char8;
-};
+    template <>
+    struct alternate_type<xlang_char8>
+    {
+        using result_type = char16_t;
```

Comment:
nit, isn't standard C++ naming convention to simply call this "type" not "result_type"?